### DISCUSSION TITLE
EU Data Residency

### DISCUSSION THREAD

**Discussion by gauthamcs** [getsentry/sentry#49564] [Launch Feedback]
## Our EU instance is generally available!

Our new data storage location in Germany is now generally available (GA). This allows all Sentry organizations— including those on our free Developer plan—  to host their data in the EU or in the US.

To learn more, see our [FAQ](https://help.sentry.io/product-features/data-residency/multi-region-support/) and our [docs](https://docs.sentry.io/product/accounts/choose-your-data-center/). 

Here's a blog about the architecture and design choices we made along the way : [A $3,000,000 Dropdown](https://sentry.engineering/blog/3m-dollar-dropdown) 

This is a significant milestone for Sentry and we’re excited to bring this critical capability to serve our EU users better. Thank you all for  your support and feedback over the past months!

Feedback or questions? Please use this Github discussion or reach out to us with on [Discord](https://discord.com/invite/sentry). 

----------------
We’ve started onboarding customers who signed for our EA on Mar 21.

* If you signed up for EA, you’ll receive an email with onboarding instructions. 

* If you've not received an email yet, appreciate your patience for just a little longer! We’ve had an overwhelming response for the EA and we’re doing a phased rollout over the coming week.
* We'll update here when we complete EA rollout. If you haven't received an email by then, please write to us at [support@sentry.io](mailto:support@sentry.io).

We’re targeting GA by April 11. 
* If you’d like to get access earlier, please use our [early access form](https://sentry.io/trust/privacy/#data-residency-form) to sign up and we'll onboard you before the GA.

Once again, thank you all for your support and patience during this journey. 

We're excited to bring you all this new capability!

__________________________________________________________________________

[Updated 18 Mar 2024]

We'll be starting EA this week and GA has moved to late March.

- We're tracking toward an EA rollout starting Mar 21, 2024. (EA rollout will likely take 3-4 business days)

- GA is slated for Mar 28, 2024

[Updated 26 Feb 2024]

We're real close now... 

- We're tracking toward an EA rollout starting Feb 29, 2024. (EA rollout will likely take 3-4 business days)

- GA is slated for Mar 18, 2024

[Updated 14 Feb 2024]

We're  _still_ in the homestretch... 

- EA rollout will start Feb 27, 2024

- GA is slated for the week of Mar 4, 2024

[Updated 26 Jan 2024]

We're in the homestretch! 

- EA rollout will start Jan 31, 2024

- GA is slated for the week of Feb 12, 2024

[Updated 05 Dec 2023]

- Early access (EA) rollout for EU data residency has been delayed by a month. We're making performance and resilience improvements based on feedback from our beta cohort.

- EA will start in mid-January 2024 [Sign up here for early access](https://sentry.io/trust/privacy/#data-residency-form)

- GA is slated for late January 2024

[Updated: 21 Nov 2023]

- Early access (EA) rollout for EU data residency will start in early December [Sign up here for early access](https://sentry.io/trust/privacy/#data-residency-form).

- We're targeting general availability (GA) in mid-December. 

- Read more about what we built on our engineering blog : [A $3,000,000 Dropdown](https://sentry.engineering/blog/3m-dollar-dropdown) 

[Updated: 9 Oct 2023]

- Early access (EA) rollout for EU data residency will start ~November 6, 2023~ in early December [Sign up here for early access](https://sentry.io/trust/privacy/#data-residency-form).

- We're targeting general availability (GA) in December. 

[Updated:  22 Sep 2023]

We’re excited to share indicative timelines for the EU data residency rollout!

- At this time, we’re targeting production rollout of EU data residency starting ~mid-October~ November 6, 2023 with our early access (EA) program followed by general availability (GA) in ~November~ December. 

- The EA will be fully functional and ready for production workloads. We will onboard customers starting the week of October 16. If you’d like to join the waitlist for early access, please sign up using [this link](https://sentry.io/trust/privacy/#data-residency-form).

We'll share further updates here and we're looking forward to any feedback. Please use this discussion thread or write to us at [support@sentry.io](mailto:support@sentry.io).
_______________________________________________________________________________________________________

Support for EU data residency has been a long standing request from our users and we’re happy to share an update on this : we will be launching an EU region later this year!

While Sentry is already compliant with data protection laws and regulations applicable to its service including GDPR, this capability allows users with European data residency requirements to store their data in Europe. 

Here’re some high-level details on what we’re building.

## What data will be stored in the EU?
When creating a Sentry organization, customers will have the option of choosing either the US or the EU as a region for that organization. Once a region is chosen, the following data will only be stored within the selected region:

- Error events, activity and issue links
- Transactions
- Profiles
- Cron check-ins
- Project metadata
- DSN keys
- Organization settings, configuration and teams
- Release health
- Releases, Debug Symbols and source maps
- Detailed Usage Data
- Session Replays
- Backups for these resources

## What data will be stored in the US?
For all organizations, the following data will be stored in the US:

- User accounts, Notification settings, and 2FA authenticators - User accounts are shared by many organizations.
- Organization Integration Metadata - Integrations can be shared across organizations, and many of our integrations only support sending their webhooks to a singular destination.
- Access Tokens for users and organizations
- Sentry Applications
- SSO, SAML and SCIM endpoints
- Organization Audit Logs

Additionally, metadata that allows Sentry to locate which region an Organization is located in, and which Organizations a User belongs to will be replicated out of the Organization’s region to facilitate login, and backwards compatible APIs.

## Will EU data residency be available on all Sentry plans?  
All plans (including our free Developer plan) will allow an Organization to be created in the EU region.

## Where in the EU will Sentry be hosting its data?
Our primary hosting site will be in Frankfurt, Germany. Sentry EU data at rest will remain in the EU.

## Can I move my existing organization to the EU region?
We will not have this capability at launch. This is on our backlog but we do not have a timeline at this point.

## When is this launching?
We’re working toward general availability in late 2023. We’ll share updates on this discussion.

## We want to hear from you!
This is a significant milestone for Sentry and we’re excited to bring this critical capability to serve our EU users better. We’d love to hear from you and share any further context on this. Please use this discussion thread or write to us at support@sentry.io.

**Comment by mattjohnsonpint** [comment]
This is fantastic news, and great timing.  I've been talking with potential Sentry users today at the NDC Oslo conference, and EU data residency has been one of the main requests.  Happy we have a time line to share. Thanks!

**Comment by johannesnagl** [comment]
please make "move existing org to EU region" part of the first launch! would be super helpful!

**Reply by gauthamcs** [reply]
@johannesnagl : Thanks for your input! We're scoping out this feature and its slated for after launch but we'll communicate here if that changes.

**Reply by nimarb** [reply]
FYI @gauthamcs , it's not possible to write to the email above. Probably "posting to this group from external" is not activated in the google group settings.

Really looking forward to you launching EU hosting. If there is a beta or similar to join, test and give feedback, we'd be happy to do so.

**Reply by gauthamcs** [reply]
@nimarb : Thanks for your note. We'll share details on the beta here and we'll definitely take you up on your offer! (I've also reached out to you on email)

**Reply by Dhrumil-Sentry** [reply]
@chadwhitacre will your org migration tool help with this use case? 
 
Noticed your discussion for the relocation tool at https://github.com/getsentry/sentry/discussions/61450

**Reply by chadwhitacre** [reply]
Likely yes, thanks for making the connection @Dhrumil-Sentry! :)

@johannesnagl @nimarb Feel free to drop me a line on #61450 or chadwhitacre@sentry.io, I'll also follow up with @gauthamcs on the state of any email conversation you're already in. 🙏

**Comment by jdenquin** [comment]
We are also very interested at Lago to get sentry for our EU cloud!

**Comment by dbartholomae** [comment]
@gauthamcs It seems that the email address above no longer exists, I get a message that I'm not allowed to send to it. We are currently preparing to start with Sentry, and the only thing that blocks us is that we would need to be on the EU cloud. Any way for us to get onto the beta?

**Reply by gauthamcs** [reply]
@dbartholomae :  Thanks for your interest! We look forward to onboarding you to Sentry. Do you mind mailing that alias again? (fixed settings and tested it works) Alternately, you could mail me directly at gautham dot chundi at sentry dot io

**Comment by sandstrom** [comment]
My feedback would be that you should make sure you setup a legal entity in EU, that we can contract with for GDPR sub-processor stuff.

Preferably in Germany or another country with high data security standards (it will seem more trustworthy than an Irish or Luxembourg subsidiary).

**Reply by gauthamcs** [reply]
@sandstrom : Good feedback and very helpful context - thank you! At this point, we have no plans for a legal entity in the EU but we'll update here if that changes.

**Reply by sandstrom** [reply]
I know it may feel like a small thing, but during sub-processor reviews that our customer performs, they care a lot about whether we've signed the contract with Microsoft Ireland or Microsoft US.

I agree it's a bit silly, but that's how it is 😄

**Reply by sandstrom** [reply]
@gauthamcs Just checking in here, any updates on a legal entity in the EU?

For many larger customers of ours, this is a deal-breaker and we'd have to disable bug-tracking for them.

Also, support for a transfer from US to EU would be great.

For us, it would be fine if we could just transfer all the settings and projects, if the data would 'remain' in the US for a while that would be fine. 

I think we'd even be fine with a 'hard break', i.e. that we just lost all data in the US when the projects were transferred to the EU (though that's less desirable for sure).

Smaller data loss of a few hours would be totally fine.

**Comment by gauthamcs** [comment]
We’re excited to share indicative timelines for the EU data residency rollout! 

At this time, we’re targeting production rollout of EU data residency starting ~mid-October~ **November 6** with our early access (EA) program followed by general availability (GA) in ~November~ **December**. We will share updates on timelines here.

The EA will be fully functional and ready for production workloads. We will onboard customers starting the week of October 16. If you’d like to join the waitlist for early access, please sign up using this link: https://sentry.io/trust/privacy/#data-residency-form.

**Comment by silpol** [comment]
any chance to have everything, like **everything**, in Europe? if not - what stops you?

**Reply by sandstrom** [reply]
What they have proposed is fine! 

It's the customer data that's interesting. Account-level data can stay in the US.

**Reply by gauthamcs** [reply]
@silpol: Great question. It's not so much what stopped us as what we were solving for - we optimized for a seamless user experience and this informed our design choices. Soon after the launch, we'll be publishing a series of tech blogs with more context on our architecture and what drove our decisions. Stay tuned!

**Reply by silpol** [reply]
@sandstrom & @gauthamcs I asked that question not for play of imagination - my current customer has batch of their customers somewhat sensitive about _any_ data stored in US. Plus if you remember dramatic development in US politics between 2016-2020, and their actor-in-chair-of-president broadcasting state secrets to various questionable parties, one must be naively-risky to keep any data over there. This is why my customer decided to stick with self-hosted and not switch to SaaS-run entity. But it's your business, not mine, so lets see where it goes.

**Reply by gauthamcs** [reply]
> we'll be publishing a series of tech blogs with more context on our architecture and what drove our decisions

Here's more context on what informed our design choices : [A $3,000,000 Dropdown](https://sentry.engineering/blog/3m-dollar-dropdown)

**Reply by johannesnagl** [reply]
so happy that this is finally made public! thanks for already being in the early adopters phase for quite a while now! 

Looking forward to read the follow-up blog post about how the 3,000,000 journey paid out in the end

**Reply by Startouf** [reply]
@sandstrom Here is what people may tell you in France in the EdTech sector

> In practice, since the invalidation of the Privacy Shield by the CJEU, not only is any transfer of personal data to the United States illegal if no supplementary measures are put in place, but using a company owned by an American parent company is illegal, even if the data is hosted in the European Union. Thus, the CNIL asked higher education and research institutions in a decision of May 27, 2021, to switch from American collaborative tools used by their students to entirely European tools, in the following terms: "American laws apply to data stored by US companies outside their territory. Therefore, there is a risk of access by US authorities to stored data." "To date, the European Data Protection Board (EDPB) has not identified any additional measures that can ensure an adequate level of protection when a transfer is made to a cloud computing service provider or other subcontractors who, in the course of their services, need access to clear data or hold encryption keys, and who are subject to US laws

So the current setup with data in the EU is a progress, but does not really help in some situations

**Comment by Startouf** [comment]
Hello @gauthamcs, 
I am the CTO of a startup laureate of the FrenchTech 2030 program, we are required to move our processes and subcontractors to the EU zone by the end of November, we are currently using Rollbar. 

Unfortunately, I found out about your plans for the EU quite late. I signed up a few days ago for the early access, but I was wondering if you would consider putting us higher in the waiting queue ? Otherwise we'd have to resort to self hosting and this is something I'd like to avoid.

I also back the previous comment from @silpol : it would be better if we could sign a contract directly with a EU entity and if everything could be handled via the EU, but I hope we can persuade our customers that this is at least a temporary measure that they should accept. Often we have to deal with people who know little about security, so it's really helpful to have those extra steps.

Let me know if it's possible to move us at the front of the waiting list, I believe we could become your ambassadors on the French side ! I believe I have signed up using "My Job Glasses" as company name (or dash-case / snake_case, I don't remember which validations I had come across ^^")

**Reply by gauthamcs** [reply]
@Startouf : Thanks for reaching out directly. We're tracking toward a late November rollout for our EA cohort and we should be able to provision your org within first few days.

**Comment by sem4phor** [comment]
any estimate when the early access rollout is distributed to all registered companies? 🙏

**Reply by gauthamcs** [reply]
@sem4phor : We're tracking toward a late November rollout for the EA registrants. Appreciate your patience!

**Reply by jkytomak** [reply]
Any news if the early access has now started or when it will start?

**Reply by gauthamcs** [reply]
@jkytomak : We're tracking for a late January EA rollout - we're almost there! We'll share any other updates on timelines here. Thank you for your support!

**Reply by jkytomak** [reply]
Has the timetable changed? We haven't got any message about EA, so do we have to wait for GA?
We need the EU region before we can take Sentry in to use in our production environment...

**Reply by johannesnagl** [reply]
also waiting for an update on that

**Comment by jorissteyn** [comment]
@gauthamcs we're an EU-based company currently managing a self-hosted Sentry instance. However, we're looking to transition from self-hosting and eagerly join the EA-rollout. Hosting within the EU is a crucial requirement for us. While we await the anticipated rollout in December, we're addressing other compliance matters with your colleagues.

I'll continue monitoring this thread. Please let me know the procedure and requirements for us to sign up. I've completed the signup form, but as we currently self-host, we're not an existing customer.

**Reply by gauthamcs** [reply]
@jorissteyn : Appreciate your patience here. We're tracking to a mid-January EA rollout and we're also building tooling to help you migrate from self hosted to SaaS. More context here: [Relocating Organizations Between Sentry Instances](https://github.com/getsentry/sentry/discussions/61450) 
If you've signed up for the EA rollout using the sign up form, you'll be part of our initial rollout - there's no other procedure or requirement on your end.

**Reply by gauthamcs** [reply]
@jorissteyn : We're missing a couple of details. What's the best email to reach you at? Alternately, you could write to us at [feedback-data-residency@sentry.io](mailto:feedback-data-residency@sentry.io).

**Reply by jorissteyn** [reply]
I'll contact you at the address provided, thanks!

**Comment by hanselsen** [comment]
It is very unclear to me where the updates on this project are posted. It seems to me that either this thread or the blog post is a valid place.
However, we are not kept up-to-date, although many people are excited to get started with Europe data residency. Including me! 😄 

Could you please shed some light on the current status and keep us up to date with moving deadlines?

**Reply by jkytomak** [reply]
I think the official place is the description of this issue, with this latest update:
```
[Updated 14 Feb 2024]
We're still in the homestretch...
    EA rollout will start Feb 27, 2024
    GA is slated for the week of Mar 4, 2024
```
Good otherwise, but you don't get a notification about changes of description, only from new comments.

**Reply by hanselsen** [reply]
@jkytomak Where did you find this information?

**Reply by jkytomak** [reply]
From top of this same page:
https://github.com/getsentry/sentry/discussions/49564#discussion-5222438

**Reply by gauthamcs** [reply]
@hanselsen 

> many people are excited to get started with Europe data residency. Including me! 😄

Thank you for your continued interest and support! We're ironing out kinks as we near rollout - we're almost there...

> Could you please shed some light on the current status and keep us up to date with moving deadlines?

We're using the issue description to message out changes and the current date. 

@jkytomak : Good callout on the lack of notifications from changes to the description. Next time, I'll update the description and post it as a new comment. Thanks for the tip!

**Comment by gauthamcs** [comment]
**EU Data Residency Timelines**
[Updated 26 Feb 2024]

We're real close now...

- We're tracking toward an EA rollout starting Feb 29, 2024. (EA rollout will likely take 3-4 business days)

- GA is slated for Mar 18, 2024

**Comment by joaocc** [comment]
Hi. Is there any way to still apply for migration to EU datacenter (the link above doesn't seem to be working). If not, how to apply for a migration after? Thanks

**Comment by gauthamcs** [comment]
@joaocc : Yes! You can still apply for early access. The sign up link was down
briefly last week but it should work now. Please let us know if this didn't
work.

Best,
Gautham

On Thu, Feb 29, 2024 at 12:50 PM Joao C Costa ***@***.***>
wrote:

> Hi. Is there any way to still apply for migration to EU datacenter (the
> link above doesn't seem to be working). If not, how to apply for a
> migration after? Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/getsentry/sentry/discussions/49564#discussioncomment-8635636>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQIW46XRILWJJVDBX3B7EK3YV6KBPAVCNFSM6AAAAAAYKZ5YN2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMMZVGYZTM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
<https://www.notion.so/sentry/Email-Signature-specs-7d543ef317d742df93b2bb211b5167af#47cba7e80fb9483bb26df120e0b20868>Gautham
Chundi
Director of Product Management
*he/him/his*
sentry.io

<https://sentryio-assets.storage.googleapis.com/img/email-signature/sentry-logo-bar.png>
<https://sentryio-assets.storage.googleapis.com/img/email-signature/sentry-logo-bar.png>

**Reply by rGaillard** [reply]
@gauthamcs we have apply for early access few month ago but we don't received any access to it yet...

**Reply by jorissteyn** [reply]
Eagerly waiting too!

**Comment by gauthamcs** [comment]
We'll be starting EA this week and GA has moved to late March.

- We're tracking toward an EA rollout starting Mar 21, 2024. (EA rollout will likely take 3-4 business days)

- GA is slated for Mar 28, 2024

**Reply by NRS-LeonBernard** [reply]
we need this eu feature. please make it happen 👍🏻

**Comment by gauthamcs** [comment]
We’ve started onboarding customers who signed for our EA on Mar 21.

If you signed up for EA, you’ll receive an email with onboarding instructions.

If you've not received an email yet, appreciate your patience for just a little longer! We’ve had an overwhelming response for the EA and we’re doing a phased rollout over the coming week.

We'll update here when we complete EA rollout. If you haven't received an email by then, please write to us at [feedback-data-residency@sentry.io](mailto:feedback-data-residency@sentry.io).

We’re targeting GA by April 11.

If you’d like to get access earlier, please use our [early access form](https://sentry.io/trust/privacy/#data-residency-form) to sign up and we'll onboard you before the GA.
Once again, thank you all for your support and patience during this journey.

We're excited to bring you all this new capability!

**Comment by GertVil** [comment]
Quick question: will the current version of https://registry.terraform.io/providers/jianyuan/sentry/latest work with the EU instance of  sentry? Do you know if this is currently supported?

**Reply by rodolfoBee** [reply]
The terraform registry for Sentry was not developed and is not maintained by Sentry. We cannot guarantee it will work with EU instances.

**Reply by GertVil** [reply]
I am assuming that there will be a different API URL for the EU instance. Since it also supports on-prem, I am guessing that it might not be a problem then. But to be sure, I will have to wait until we have received our EA access to test this. Thank you!

**Reply by gauthamcs** [reply]
@GertVil : Thanks for raising this. We'll get back to you on EU support for the Terraform provider.

**Reply by joaocc** [reply]
@gauthamcs just for reference, in the last few weeks, re-applying modules that used that provider resulted in changes to the URL. The changes included a new infix ".us" and then changing order of 2 DNS elements inside.
This seems to indicate that the API is returning different endpoints to the provider.

**Comment by NoSync** [comment]
This is fantastic news, but I'd like a clarification. In your FAQ page, you state:

> Q: Can I move my existing Sentry organization to the EU region?
> A: Unfortunately, we do not have a mechanism for customers to perform a US —> EU relocation.

Whereas in this conversation you say:

> Can I move my existing organization to the EU region?
> We will not have this capability at launch but this is on our backlog.

I understand it's not possible to currently migrate an organization from the US to the EU, and that's fine. But will this feature actually be added at some point in the future, or has the idea been abandoned?

Thanks.

**Reply by gauthamcs** [reply]
@NoSync : Thanks for calling this out. This capability is on our backlog but it's unlikely we'll get to this over the coming months. We'll update this thread if that changes. I've also update this discussion to reflect this.

**Reply by MaximeLaurenty** [reply]
For those like me who wonder: the "Transfer project" feature in the project settings page doesn't work cross-region for now (which was expectable given Gautham's message above).

**Reply by mathiasose** [reply]
@gauthamcs @mikejihbe The article at https://sentry.engineering/blog/3m-dollar-dropdown also claims this is "coming soon", maybe it should be clarified.
![image](https://github.com/user-attachments/assets/04545125-012e-4c62-b29b-5a4cd5fce639)

I understand that creating a tool to migrate data across regions on demand is tricky (and that re-creating your org and transferring the subscription is the current best solution), but do you have any issue tracker for progress on this tool that we can subscribe to updates for, or a signup list to show interest?

**Reply by xkabylgSICKAG** [reply]
@gauthamcs is that migration tool still in the backlog? We would prefer to migrate our data to EU, but losing historical data is something we'd prefer to avoid if possible. Should we bite the bullet and just change to EU anyway or is it worth waiting for this migration tool to be available? We're not in a rush.

**Reply by sandstrom** [reply]
We are also waiting for a migration tool, but would be 100% with only migrating settings, and no data. 

Either a hard-cutoff where we loose all data would be totally fine, or some ability to login to the US datacenter during an interim period, to see the old data.

**Reply by gauthamcs** [reply]
@xkabylgSICKAG : We don't have a timeline on the migration tool and I'd recommend not waiting on this. We looked into what it'd take to enable this capability at scale and this would be non-trivial. At this time, we're not prioritizing a migration tool that'd include data.

@sandstrom : 

> or some ability to login to the US datacenter during an interim period, to see the old data.

For clarity, you want access to your US org and its data for an interim period after you migrate to our EU region? I'll confirm if we can enable this. Could you also write in to support@sentry.io referencing this comment?

**Reply by sandstrom** [reply]
Yes, access to US org during an interim period (e.g. 30 days, after which it could be deleted). 

That would allow us to update all systems to ship errors to the new EU org, but still have access to old data for troubleshooting during a few weeks.

But we would also be fine with a hard cutoff, where we spin up a new EU org as a perfect copy of the US org, and then delete the US org. But this would be worse than interim access to both.

**Reply by Anwardo** [reply]
> @xkabylgSICKAG : We don't have a timeline on the migration tool and I'd recommend not waiting on this. We looked into what it'd take to enable this capability at scale and this would be non-trivial. At this time, we're not prioritizing a migration tool that'd include data.

Understand that data migration is probably not coming (anytime soon), any news on a settings migrations only though?

**Comment by sem4phor** [comment]
Hello, I got EA and switched my org slug as well as my DSN. The Sentry Auth Token should still be valid.

However my Upload of debug symbols when using expo build (react-native) is failing :/ 
Do i have to change the sentry base url too or something like that?

**Reply by markstory** [reply]
>  Do i have to change the sentry base url too or something like that?

You may need to update the sentry-cli version that you're using to upload symbol files. Setting a base URL to de.sentry.io can also help. Knowing more about the errors you're having would help here.

**Reply by mcuelenaere** [reply]
We are seeing the same issue. We upgraded to sentry-cli 2.31.0 and are seeing a HTTP 401 (`Invalid org token`) when using sentry.io (which is the wrong endpoint) and the `error: Two different url values supplied: https://sentry.io (from token), https://de.sentry.io/.` error when using de.sentry.io (which is the correct endpoint).

**Reply by lsmith77** [reply]
I installed `sentry-cli 2.31.0` via homebrew and it works but it keeps outputting `Unrecognized auth token format!`.

it looks like `sntryu_[..]`, where `[..]` is a 64 char string containing lower case letters and numbers.

**Reply by mcuelenaere** [reply]
> it looks like `sntryu_[..]`, where `[..]` is a 64 char string containing lower case letters and numbers.

Isn't that an user auth token? I believe you need to use an organization auth token.

**Comment by lsmith77** [comment]
I created a new org, but I am unable to start a trial to finalize the setup (ie. setup integrations).

**Reply by markstory** [reply]
What happens when you try to start the trial? I'm not seeing any errors from any organizations in the de region. If you could share your org slug either privately or in a reply that would be helpful in getting your trial sorted.

**Reply by lsmith77** [reply]
<img width="280" alt="image" src="https://github.com/getsentry/sentry/assets/300279/ba534908-a0f2-4692-b9c3-e5736431e7a9">

**Reply by lsmith77** [reply]
`witty-works-ag`

**Reply by lsmith77** [reply]
any update here @markstory ?

**Reply by lsmith77** [reply]
ok .. it appears it was activated after all and I am down to 9 days on the trial.

I managed to connect slack but github when I go to configure, I have already given sentry permission on the US instance and then I doesn't seem to be able to also grant permissions to the EU instance.

**Reply by lsmith77** [reply]
I have opened https://github.com/getsentry/sentry/issues/68780

**Comment by jkytomak** [comment]
Got EA and created new org in EU.
Server monitoring works fine, but client (Next.js) not.
Client gets from `/api/monitoring` requests this response:
```
403 Forbidden
{"detail":"event submission rejected with_reason: ProjectId"}
```
If I change the client dsn to old US one, client monitoring works again.

Ps. we are using the same org slug as previously in US (renamed US one before creating EU)

**Reply by jkytomak** [reply]
Tried also with totally new slug, had no effect, so not related to that at least.

**Reply by jkytomak** [reply]
Created an issue:
https://github.com/getsentry/sentry/issues/68679

**Comment by gauthamcs** [comment]
Update: We’re targeting GA by April 16.

If you’d like to get access earlier, please use our [early access form](https://sentry.io/trust/privacy/#data-residency-form) to sign up and we'll onboard you before the GA.

**Comment by gauthamcs** [comment]
Our new data storage location in Germany is now generally available (GA). This allows all Sentry organizations— including those on our free Developer plan— to host their data in the EU or in the US.

To learn more, see our [FAQ](https://help.sentry.io/product-features/data-residency/multi-region-support/) and our [docs](https://docs.sentry.io/product/accounts/choose-your-data-center/).

Here's a blog about the architecture and design choices we made along the way : [A $3,000,000 Dropdown](https://sentry.engineering/blog/3m-dollar-dropdown)

This is a significant milestone for Sentry and we’re excited to bring this critical capability to serve our EU users better. Thank you all for your support and feedback over the past months!

Feedback or questions? Please use this Github discussion or reach out to us with on [Discord](https://discord.com/invite/sentry).

**Comment by joaocc** [comment]
Can you please confirm that there is no way to migrate existing accounts (older ones) to EU region (as in, without recreation)? Thx

**Reply by johannesnagl** [reply]
correct. we already migrated and it took us 1,5 business days to set up everything again. painful, but at least our GDPR team is happy in the end.

**Reply by gauthamcs** [reply]
@joaocc : There isn't a mechanism to migrate an existing org to the EU region - you'll have to recreate it. 
However, you can transfer your subscription to the EU. See this reply (https://github.com/getsentry/sentry/discussions/49564#discussioncomment-9167369)

**Comment by lsmith77** [comment]
Is the expectation that we open a new subscription on the EU instance even if our US subscription is still running for several months? Or is there some process by which we can at least get a discount in such cases or better yet move the subscription?

**Reply by gauthamcs** [reply]
@lsmith77 : Yes, you can transfer your subscription to the your new EU org. We’ll cancel your subscription plan for your US organization and apply any remaining credit to your new EU org. 

To transfer your subscription:
- Open a new subscription on the EU instance 
- Write in to support@sentry.io with the subject “Billing Transfer to EU Org”. Include your new org slug, your current org slug and your desired transfer date. 

We'll be documenting this on our [Help Center](https://help.sentry.io/product-features/data-residency/multi-region-support/).

Updated: See Help Center article [Can I transfer my US Region Sentry subscription to the EU Region?](https://sentry.zendesk.com/hc/en-us/articles/25078328002075-Can-I-transfer-my-US-Region-Sentry-subscription-to-the-EU-Region)

**Comment by gauthamcs** [comment]
Help Center article on transferring subscription from your US org to your new EU org : [Can I transfer my US Region Sentry subscription to the EU Region?](https://sentry.zendesk.com/hc/en-us/articles/25078328002075-Can-I-transfer-my-US-Region-Sentry-subscription-to-the-EU-Region)

**Comment by caledancho** [comment]
Hi Gautham- Looks like this was sent via automation on behalf of Cale, but
he's no longer with Hashicorp. However, I'm cc'ing Nic (product manager for
data residency) to weigh in on Sentry's value for our Europe customers. If
needed, perhaps the two of you can connect about this.

Bryan

**Comment by sachingharge** [comment]
@gauthamcs Is there any plan to support in place change of data residency location instead of creating new org and transferring data to new org?

**Reply by markstory** [reply]
There aren't any plans to support in-place region relocations at this point in time.

**Comment by tts-sdrissen** [comment]
Germany has two cities named Frankfurt. 

Are the servers in 
- Frankfurt am Main (State of Hesse)
or
- Frankfurt an der Oder (State of Brandenburg)
?

**Reply by MaximeLaurenty** [reply]
I believe all AWS/GCP/Azure regions in "Frankfurt" are in Frankfurt am Main. So it's quite safe to assume Sentry is hosted in Frankfurt am Main as well.
